lib/widgets/app.dart

```diff
@@ -268,41 +266,50 @@ class ChooseAccountPage extends StatelessWidget {
         // The default trailing padding with M3 is 24px. Decrease by 12 because
         // IconButton (the "…" button) comes with 12px padding on all sides.
         contentPadding: const EdgeInsetsDirectional.only(start: 16, end: 12),
-        onTap: () => Navigator.push(context,
-          HomePage.buildRoute(accountId: accountId))));
+        onTap: () => HomePage.navigate(context, accountId: accountId)));
   }
 
   @override
   Widget build(BuildContext context) {
+    final colorScheme = ColorScheme.of(context);
     final zulipLocalizations = ZulipLocalizations.of(context);
     assert(!PerAccountStoreWidget.debugExistsOf(context));
     final globalStore = GlobalStoreWidget.of(context);
-    return Scaffold(
-      appBar: AppBar(
-        title: Text(zulipLocalizations.chooseAccountPageTitle),
-        actions: const [ChooseAccountPageOverflowButton()]),
-      body: SafeArea(
-        minimum: const EdgeInsets.fromLTRB(8, 0, 8, 8),
-        child: Center(
-          child: ConstrainedBox(
-            constraints: const BoxConstraints(maxWidth: 400),
-            child: Column(mainAxisSize: MainAxisSize.min, children: [
-              Flexible(child: SingleChildScrollView(
-                padding: const EdgeInsets.only(top: 8),
-                child: Column(mainAxisSize: MainAxisSize.min, children: [
-                  for (final (:accountId, :account) in globalStore.accountEntries)
-                    _buildAccountItem(context,
-                      accountId: accountId,
-                      title: Text(account.realmUrl.toString()),
-                      subtitle: Text(account.email)),
-                ]))),
-              const SizedBox(height: 12),
-              ElevatedButton(
-                onPressed: () => Navigator.push(context,
-                  AddAccountPage.buildRoute()),
-                child: Text(zulipLocalizations.chooseAccountButtonAddAnAccount)),
-            ]))),
-      ));
+    // Borrowed from [AppBar.build].
+    // See documentation on [ModalRoute.impliesAppBarDismissal]:
+    /// > Whether an [AppBar] in the route should automatically add a back button or
+    /// > close button.
+    final hasBackButton = ModalRoute.of(context)?.impliesAppBarDismissal ?? false;
```

Comment:
nit: set off this stanza with blank lines:
```suggestion

    // Borrowed from [AppBar.build].
    // See documentation on [ModalRoute.impliesAppBarDismissal]:
    /// > Whether an [AppBar] in the route should automatically add a back button or
    /// > close button.
    final hasBackButton = ModalRoute.of(context)?.impliesAppBarDismissal ?? false;

```

In particular that helps make clear what code the comment is describing.

Reply:
nit: those `///` should be `//`, right?